### PR TITLE
feat: support scoped commits

### LINKED ISSUE #31: Still didn't work
Tried 1.6.2 and it didn't seem to work I had a line (feat(play): test) which should have generated notes for that and it didnt post anything to the PR.

```
processing line: ea19f82 feat(play): test
adding "test (ea19f82)" to others
```

### PR DESCRIPTION
**Pull request by ghost**
Fixes #31

Currently the `prefix` can contain both `type` and `scope` but it's treated as `type` only. So I've added a step to parse those out. The scope is used in the line description similar to how they show up in conventional release notes.

I had to include `tslib` as to target ES2018 and use various syntax (like destructuring), it's needed.

```
src/commit.ts:75:15 - error TS2354: This syntax requires an imported helper but module 'tslib' cannot be found.

75       const { type, scope } = getTypeAndScope(words[1])
```

Downgrading the target to ES2015 avoids the error. I guess they include helpers automatically in ES2015 and stopped doing that in later versions.

I've also committed changes in the `dist` folder as that seems to be convention here.

Since the tsbuildinfo is on `3.8.3`, I made a separate commit to upgrade to 3.9.9 which allows us to use the latest `tslib`. I can split this into a separate PR if needed.

### CODE REVIEW

File: dist/index.js [hunk 1]
```diff
@@ -1299,6 +1299,7 @@ module.exports = {
 "use strict";
 
 Object.defineProperty(exports, "__esModule", { value: true });
+exports.previewFromCommits = void 0;
```
Comment (ghost) [inline dist/index.js:1302]: TS 3.9.9 output change. It defines the export first then assigns it later.

File: src/commit.ts [hunk 2]
```diff
@@ -27,6 +27,20 @@ async function commitHeadersSinceVersion(version: string): Promise<string[]> {
   return commitsSinceLastVersionOutput.stdout.trim().split('\n');
 }
 
+function getTypeAndScope(prefix: string): { type: string, scope?: string } {
+  const matches = /^([^(]*)(?:\(([^)]+)\))?:/.exec(prefix);
+  if (!matches) {
+    // Malformed commit prefix. Try our best.
+    return { 
+      type: prefix.split(':')[0] || ''
```
Comment (ghost) [inline src/commit.ts:35]: I've kept this logic from previous behaviour in case some existing user is reliant on it.

File: src/commit.ts [hunk 3]
```diff
@@ -58,13 +72,18 @@ function processCommits(commitHeaders: string[]): ICommitData {
...
   
-      const lineDescription = `${rest} (${hash})`;
-      if (prefix in processedCommits) {
-        console.log(`adding "${lineDescription}" to ${prefix}`);
-        processedCommits[prefix].push(lineDescription);
+      const lineDescription = [
+        scope ? `${scope}:` : undefined,
+        rest,
+        `(${hash})`
+      ].filter(part => !!part).join(' ');
```
Comment (ghost) [inline src/commit.ts:82]: I've split these sections up as scope is optional and a ternary inside a template literal is confusing.

**Review by Shesekino — APPROVED** [review]
Didn't review this thoroughly, but I rather not block this feature :) Thanks!

**Comment by Shesekino** [comment]
@jahed-snyk would you be able to reopen the PR from a branch on this repo rather than a fork?

**Comment by ghost** [comment]
@Shesekino will do, I assumed I didn't have write-access. Is there a testing process for this? If not, I'll try this out on a test repo before merging just to confirm everything's working.

**Comment by Shesekino** [comment]
A test repo sounds fantastic 👌
We were missing permissions on the dev group - I've modified that :) Let me know if you're missing anything please 🙏

**Review by cesar6661 — APPROVED** [review]

**Comment by ghost** [comment]
There's a good chance I won't have time for this... So closing for now. If anyone else wants to pick it up feel free to.

**Pull request closed without merging**